debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java

```diff
@@ -507,6 +511,92 @@ public static SecureConnectionMode parse(String value, String defaultValue) {
         }
     }
 
+    /**
+     * Set of predefined connector adapter modes.
+     */
+    public enum ConnectorAdapterMode implements EnumeratedValue {
+        /**
+         * Expects the target database to be MySQL using the MySQL driver.
+         * This should also be used if the target database is MySQL compliant but isn't MariaDB.
+         */
+        MYSQL("mysql") {
+            @Override
+            protected ConnectorAdapter getAdapter(MySqlConnectorConfig connectorConfig) {
+                LOGGER.info("Using " + MySqlConnectorAdapter.class.getName());
+                return new MySqlConnectorAdapter(connectorConfig);
+            }
+        },
+
+        /**
+         * Expects the target database to be MariaDB using the MariaDB driver.
+         */
+        MARIADB("mariadb") {
+            @Override
+            protected ConnectorAdapter getAdapter(MySqlConnectorConfig connectorConfig) {
+                LOGGER.info("Using " + MariaDbConnectorAdapter.class.getName());
+                return new MariaDbConnectorAdapter(connectorConfig);
+            }
+        },
+
+        /**
+         * Expects the target database to be MariaDB but uses the MySQL driver.
+         */
+        MARIADB_HYBRID("mariadb-hybrid") {
```

Comment:
Is there any advanatage in this mode? I thought it to be a transient one till MariaDB db + MariaDB driver is in place which already is.

Reply:
I'm on the fence about it.  

I imagine someone upgrading from an older version who may want to use the legacy behavior, and that's partly why I left it.  If anything, I think it would be something we would leave in for one minor release and remove to give people a chance to adapt with a buffer.  wdyt?

Reply:
Sounds reasonable. Maybe we should add to the documentation that this is not preferred mode of opertion or something like that.